src/backend/aspen/covidhub_import/import_projects.py

```diff
@@ -234,6 +234,27 @@ def import_project(
             )
         }
 
+        # load the existing samples, store as a mapping between public id to
+        # sample.
+        public_id_to_samples: Mapping[str, Sample] = {
+            sample.public_identifier: sample
+            for sample in (
+                session.query(Sample)
+                .filter(Sample.submitting_group == group)
+                .options(
+                    joinedload(Sample.uploaded_pathogen_genome)
+                    .joinedload(
+                        UploadedPathogenGenome.consuming_workflows.of_type(  # type: ignore
+                            GisaidAccessionWorkflow
+                        )
+                    )
+                    .joinedload(
+                        GisaidAccessionWorkflow.outputs.of_type(GisaidAccession)  # type: ignore
+                    )
+                )
+            )
+        }
+
```

Comment:
This should be combined with the immediately preceding block as one query:

```
samples = (session.query(Sample)
               .filter(Sample.submitting_group == group)
               .options(
                   joinedload(Sample.uploaded_pathogen_genome)
                   .joinedload(
                       UploadedPathogenGenome.consuming_workflows.of_type(  # type: ignore
                           GisaidAccessionWorkflow
                       )
                   )
                   .joinedload(
                       GisaidAccessionWorkflow.outputs.of_type(GisaidAccession)  # type: ignore
                   )
               )
               .all()
)
external_accessions_to_samples: Mapping[str, Sample] = {
    sample.private_identifier: sample
    for sample in samples
}
public_id_to_samples: Mapping[str, Sample] = {
    sample.public_identifier: sample
    for sample in samples
}
```

Reply:
done!